s3_resource/src/s3_resource.cpp

```diff
@@ -1830,7 +1830,8 @@ irods::error s3PutCopyFile(
             s3_logger::error( msg );
             return ERROR( SYS_MALLOC_ERR, msg );
         }
-        g_mpuData = (multipart_data_t*)calloc(totalSeq, sizeof(multipart_data_t));
+        //g_mpuData = (multipart_data_t*)calloc(totalSeq, sizeof(multipart_data_t));
+        g_mpuData = new multipart_data_t[totalSeq];
```

Comment:
Looks like there's some early returns which could lead to a memory leak.

Given that `calloc` was used before this change, I think it's okay. I'll open an issue for this.

Reply:
Created #2256.